addons/explore-editor-theme/ui/IconSaver.gd

```diff
@@ -0,0 +1,34 @@
+tool
+extends VBoxContainer
+
+
+signal filesystem_changed()
+
+export var icon_name := ''
+export var type_name := ''
+
+onready var _save_dialog: FileDialog = $SaveDialog
+onready var _message: Label = $Message
+
+
+func clear_message() -> void:
+	_message.text = ''
+
+
+func show_save_dialog() -> void:
```

Comment:
Consider naming signal handlers here and below in the similar manner to other classes, i.e. `_on_show_save_pressed`. Key points of this naming scheme: `_` for "private" and `_on_verbed` for semantics.

Reply:
I personally consider them public as there is no harm from calling them from somewhere else, but I can also see why they are kinda private. Will change them to `_to_verbed` equivalents

Reply:
Though in some cases they don't have any parameters so it'd be easy to expose them, often signal handlers rely on some data being passed from the originator of the signal. In those cases if I'd like to expose something, I'd just make a new method that has safe to use signature and that internally calls the same callback.